src/util.cpp

```diff
@@ -418,6 +418,7 @@ bool SoftSetArg(const std::string& strArg, const std::string& strValue)
     if (mapArgs.count(strArg))
         return false;
     mapArgs[strArg] = strValue;
+    _mapMultiArgs[strArg].push_back(strValue);
```

Comment:
We do not take the lock when accessing mapMultiArgs all over the codebase. You cannot write a new entry to this map without updating the various locations in the codebase which access mapMultiArgs to also go through cs_args.